pkg/engine/armvariables_test.go

```diff
@@ -390,6 +411,21 @@ func TestK8sVars(t *testing.T) {
 		t.Errorf("unexpected diff while expecting equal structs: %s", diff)
 	}
 
+	// Test with HostedMaster + MSI
+	cs.Properties.OrchestratorProfile.KubernetesConfig.UseManagedIdentity = true
+	varMap, err = GetKubernetesVariables(cs)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	expectedMap["useManagedIdentityExtension"] = "true"
```

Comment:
This AKS ("HostedMaster") UT reflects that when an AKS configuration includes the UseManagedIdentity = true setting we *only* set the `useManagedIdentityExtension` variable to `"true"`, but we *do not* change the `servicePrincipalClientId` and `servicePrincipalClientSecret` variables to `"msi"`. In other words, this section of the UT proves that @norshtein's change is doing what we want it to do, and that it will be protected by UT over time.